onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -27,12 +27,18 @@ class ReduceKernelBase {
       ORT_ENFORCE(info.GetAttr("keepdims", &keepdims).IsOK());
     }
     keepdims_ = (keepdims == 1);
+    int64_t noop_with_empty_axes = info.GetAttrOrDefault<int64_t>("noop_with_empty_axes", 0);
+    noop_with_empty_axes_ = (noop_with_empty_axes == 1);
+    if (axes_.size() > 0) {
+      ORT_ENFORCE(noop_with_empty_axes_ == false, "Noop when axes is not empty is not allowed.");
```

Comment:
The value of axes_ will not be known when the graph is built with the attribute 'noop_with_empty_axes', so we cannot enforce this condition.